### ISSUE TITLE
[cm12.1_golden.nova.20151023] Cannot switch between 2G/3G

### ISSUE THREAD

**Issue by treblada** [NovaFusion/android_device_samsung_golden#64]
On my S3mini it's impossible to enforce 2G/3G usage, since the update from Nova's cm11.
I followed the installation procedure and wiped the phone completely before installing  cm12.

What I tried:
1. In the Setting/Mobile Networks the "Preferred network type" I set to 3G, nevertheless the status icon for radio remained showing an "E" (3G _is_ available in the area, it worked with cm11)
2. I've created two profiles ("2G" and "3G") where everything is left unchanged except for the "Preferred network" which is set to 2G resp. 3G. When I select one of those profiles, it has no impact whatsoever on the current radio type, but it _always_ switches the mobile data off.

Generally 3G is possible (it worked _once_ on cm 12.1) , but it's completely unclear to me how it got activated - I cannot reproduce this "miracle".

**Comment by luisfalonso** [comment]
Same issue on cm12.1_golden.nova.20160210... on stock 4.1.2, when i chaged network type, temporaly switches off/on data, but network type is changed.

**Comment by luisfalonso** [comment]
Workaround:
- Go to mobile connections and disable data connection
- Wait until connection is disabled (a pair of seconds).
- Change data connection type.
- Reenable data connection.

**Comment by treblada** [comment]
Thanks for the workaround. Seem to work for me too.
But since I've tried it, my WiFi is malfunctioning. I can only connect to a network ONCE. If I deactivate wifi and reactivate it, my phone won't connect to it any more. Neither automatically, nor manualle. I have to remove the network and reconnect to it again. 100% reproducable, at home and at work.
Strange issue and possibly not related to mobile connection switching, but it started right after I've started playing with mobile data mode switching.
